synapse/module_api/__init__.py

```diff
@@ -1563,6 +1564,29 @@ async def get_monthly_active_users_by_service(
             start_timestamp, end_timestamp
         )
 
+    async def get_canonical_room_alias(self, room_id: RoomID) -> Optional[RoomAlias]:
+        """
+        Retrieve the given room's current canonical alias.
+
+        A room may declare an alias as "canonical", meaning that it is the
+        preferred alias to use when referring to the room. This function
+        retrieves that alias from the room's state.
+
+        Added in Synapse v1.84.0.
+
+        Args:
+            room_id: The Room ID to find the alias of.
+
+        Returns:
+        None if the room ID does not exist, or if the room exists but has no canonical alias.
+        Otherwise, the parsed room alias.
```

Comment:
```suggestion
        Returns:
            None if the room ID does not exist, or if the room exists but has no canonical alias.
            Otherwise, the parsed room alias.
```